File: parciales/ED1-Parcial1-2021-1--002/codigo/RPN.java
import java.util.LinkedList;
class RPN{

 public static int solve(String[] tokens){
  String operators = "+-*/";
  LinkedList<String> aList = new LinkedList<String>();
  for(String t : tokens){
    if(!operators.contains(t)){
      aList.add(0,t);
    }else{
      int a = Integer.valueOf(aList.get(0));
      aList.remove(0);
      int b = Integer.valueOf(aList.get(0));
      aList.remove(0);
      int index = operators.indexOf(t);
      if(index == 0) 
       aList.add(0,String.valueOf(a+b));
      else if(index == 1)
       aList.add(0,String.valueOf(b-a));
      else if(index == 2)
       aList.add(0,String.valueOf(a*b));
      else
       aList.add(0,String.valueOf(b/a));
    }
  }
  String last = aList.get(0);
  return Integer.valueOf(last);
  }
 
  public static void main(String[] args){
    String[] arr = {"3","10","5","+","*"};
    System.out.println(solve(arr));
  }

}

File: parciales/ED1-Parcial1-2021-1--002/codigo/RPN.py
def solve(tokens):
  operators = "+-*/"
  aList = []
  for t in tokens:
    if not t in operators:
      aList = [t] + aList
    else:
      a = int(aList[0])
      aList = aList[1:]
      b = int(aList[0])
      aList = aList[1:]
      index = operators.index(t)
      if index == 0:
       aList = [str(a+b)] + aList 
      elif index == 1:
       aList = [str(a-b)] + aList 
      elif index == 2:
       aList = [str(a*b)] + aList 
      else:
       aList = [str(b/a)] + aList 
  last = aList[0]
  return int(last)

def __main__():
  print(solve(["3","10","5", "+", "*"]))

__main__()

